callreq/src/data.rs

```diff
@@ -151,19 +168,35 @@ impl<Q: Display + FromStr> FromStr for CallScope<Q> {
 
     fn from_str(s: &str) -> Result<Self, Self::Err> {
         match ContractId::from_str(s) {
-            Err(err1) => {
-                let s = s.trim_start_matches("contract:");
-                let query = Q::from_str(s).map_err(|_| err1)?;
-                Ok(Self::ContractQuery(query))
+            Err(err_contract_id) => {
+                if let Some(query_str) = s.strip_prefix("contract:") {
+                    Q::from_str(query_str)
+                        .map(CallScope::ContractQuery)
+                        .map_err(|_| err_contract_id)
+                } else {
+                    Err(err_contract_id)
+                }
```

Comment:
We have failing lints, and one of the reasons is this:

```suggestion
                let query_str = s.strip_prefix("contract:").ok_or(err_contract_id)?;
                Q::from_str(query_str)
                    .map(CallScope::ContractQuery)
                    .map_err(|_| err_contract_id)
```